File: NeuralNetNative/SqrtReverseFactor.cpp
#define NEURALNETNATIVEAPI
#include <mathimf.h>
#include "SqrtReverseFactor.h"

namespace NeuralNetNative {
	float SqrtReverseFactor::GetFactor(int iterNumber) {
		const float Epsilon = 0.000001f;
		return invsqrtf(iterNumber + Epsilon);
	}
}

File: NeuralNet/Train/LeanFactorStrategy/SqrtReverseFactor.cs
using System;

namespace NeuralNet.LeanFactorStrategy {
	public sealed class SqrtReverseFactor : ILearnFactorStrategy {
		private const float Epsilon = 0.000001f;
		
		public float GetFactor(int iterNumber) {
			return (float) Math.Sqrt(1.0/(iterNumber + Epsilon));
		}
	}
}
